### PR TITLE
luminous: mon: osd feature checks with 0 up osds

### PR DESCRIPTION
**Pull request by kmroz**
http://tracker.ceph.com/issues/21544

### CODE REVIEW

File: src/mon/MonCommands.h [hunk 1]
```diff
@@ -739,13 +739,15 @@ COMMAND("osd erasure-code-profile ls", \
...
-	"name=key,type=CephChoices,strings=full|pause|noup|nodown|noout|noin|nobackfill|norebalance|norecover|noscrub|nodeep-scrub|notieragent|sortbitwise|recovery_deletes|require_jewel_osds|require_kraken_osds", \
+	"name=key,type=CephChoices,strings=full|pause|noup|nodown|noout|noin|nobackfill|norebalance|norecover|noscrub|nodeep-scrub|notieragent|sortbitwise|recovery_deletes|require_jewel_osds|require_kraken_osds " \
+	"name=sure,type=CephChoices,strings=--yes-i-really-mean-it,req=false", \
 	"set <key>", "osd", "rw", "cli,rest")
 COMMAND("osd unset " \
 	"name=key,type=CephChoices,strings=full|pause|noup|nodown|noout|noin|nobackfill|norebalance|norecover|noscrub|nodeep-scrub|notieragent", \
 	"unset <key>", "osd", "rw", "cli,rest")
 COMMAND("osd require-osd-release "\
-	"name=release,type=CephChoices,strings=luminous",
+	"name=release,type=CephChoices,strings=luminous|mimic " \
```
Comment (liewegas) [inline src/mon/MonCommands.h]: I think we should leave off the mimic option here

File: src/mon/OSDMonitor.cc [hunk 2]
```diff
@@ -8985,9 +9025,17 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
...
+    if (rel == CEPH_RELEASE_MIMIC) {
+      if (!osdmap.get_num_up_osds() && sure != "--yes-i-really-mean-it") {
+        ss << "Not advisable to continue since no OSDs are up. Pass "
+           << "--yes-i-really-mean-it if you really wish to continue.";
+        err = -EPERM;
+        goto reply;
+      }
+      if ((!HAVE_FEATURE(osdmap.get_up_osd_features(), SERVER_MIMIC))
+           && sure != "--yes-i-really-mean-it") {
+	ss << "not all up OSDs have CEPH_FEATURE_SERVER_MIMIC feature";
```
Comment (liewegas) [inline src/mon/OSDMonitor.cc]: ...and here.  It will always be a bad idea for them to do this (via the --force option) on luminous.

**Comment by kmroz** [comment]
@liewegas - I dropped the mimic reference in MonCommands.h and the mimic block in OSDMonitor.cc, relying on the original
```c++
    if (rel == CEPH_RELEASE_LUMINOUS) {
      if (!HAVE_FEATURE(osdmap.get_up_osd_features(), SERVER_LUMINOUS)) {
	ss << "not all up OSDs have CEPH_FEATURE_SERVER_LUMINOUS feature";
	err = -EPERM;
	goto reply;
      }
    }
```

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-10-26_18:02:19-rados-wip-sage-testing-2017-10-26-1137-distro-basic-smithi/

**Pull request merged**